### ISSUE TITLE
adapter-vercel doesn't redirect prerendered pages with incorrect trailing slash

### ISSUE THREAD

**Issue by KevsRepos** [sveltejs/kit#8322]
### Describe the bug

https://svelte.dev/blog/ loads without css, but https://svelte.dev/blog works fine. Same bug appears on all subroutes: https://svelte.dev/blog/whats-new-in-svelte-january-2023/ and so on...

### Reproduction

I've just experienced it on the svelte.dev/blog site, but this looks like a kit specific error and not like a site specific error.

### Logs

```Shell
Loading module from “https://svelte.dev/blog/_app/immutable/start-e5c89509.js” was blocked because of a disallowed MIME type (“text/html”).

And the various css files couldnt load aswell.
```

### System Info

```Shell
System:
    OS: Linux 6.0 Pop!_OS 22.04 LTS
    CPU: (8) x64 Intel(R) Core(TM) i7-8565U CPU @ 1.80GHz
    Memory: 254.69 MB / 15.51 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 17.2.0 - ~/.nvm/versions/node/v17.2.0/bin/node
    Yarn: 1.22.15 - ~/.nvm/versions/node/v17.2.0/bin/yarn
    npm: 8.1.4 - ~/.nvm/versions/node/v17.2.0/bin/npm
  Browsers:
    Firefox: 108.0
```

### Severity

annoyance

**Comment by KevsRepos** [comment]
Just realized I set `annoyance` because I only experienced it on the svelte.dev website, but I guess this is an actual dealbreaker if this bug would appear on my own websites. So its more like a "cant use Sveltekit"

**Comment by Rich-Harris** [comment]
This is an issue with `adapter-vercel` — the platform doesn't differentiate between `/blog` and `/blog/` if `blog.html` is a prerendered file.

**Comment by Rich-Harris** [comment]
Looks like this was actually fixed by #8766 — closing
